src/bindings/python/flux/util.py

```diff
@@ -958,12 +958,91 @@ def load(self):
+                "format":      "format",
+              },
+              "name2": ..
+            }
+
+        Raises ValueError on failure.
+        """
+        if not isinstance(formats, dict):
+            raise ValueError(f"{path}: the 'formats' key must be a mapping")
+        for name, entry in formats.items():
+            if name in self.builtin_formats:
+                raise ValueError(
+                    f"{path}: override of builtin format '{name}' not permitted"
+                )
+            if not isinstance(entry, dict):
+                raise ValueError(f"{path}: 'formats.{name}' must be a mapping")
+            if "format" not in entry:
+                raise ValueError(
+                    f"{path}: formats.{name}: required key 'format' missing"
+                )
+            if not isinstance(entry["format"], str):
+                raise ValueError(
+                    f"{path}: formats.{name}: 'format' key must be a string"
+                )
+
     def validate(self, path, conf):
         """
         Validate config file as it is loaded before merging it with the
         configuration. This function does nothing in the base class, but
         subclasses may define it to implement higher level validation.
         """
 
+    def get_format_string(self, format_name):
+        """
+        Convenience function to fetch a format string from the current
+        config. Assumes the current config has been loaded and contains
+        a "formats" table.
+
+        Special format names:
+            help: print a list of configured formats
+            get-config=NAME: dump the config for format "name"
+
+        Args:
+            format_name: Name of the configured format to return
+        """
+        if "{" in format_name:
+            return format_name
+
+        formats = self.formats
+        if format_name == "help":
+            print(f"\nConfigured {self.name} output formats:\n")
+            for name, entry in formats.items():
+                print(f"  {name:<12}", end="")
+                try:
+                    print(f" {entry['description']}")
+                except KeyError:
+                    print()
+            sys.exit(0)
+        elif format_name.startswith("get-config="):
+            _, name = format_name.split("=", 1)
+            try:
+                entry = formats[name]
+            except KeyError:
+                raise ValueError(f"--format: No such format {name}")
+            print(f"[formats.{name}]")
+            if "description" in entry:
+                print(f"description = \"{entry['description']}\"")
+            print(f"format = \"{entry['format']}\"")
+            sys.exit(0)
+        try:
+            return formats[format_name]["format"]
+        except KeyError:
+            raise ValueError(f"--format: No such format {format_name}")
+
     def __getattr__(self, attr):
         return self.dict[attr]
+
+    def __getitem__(self, item):
+        return self.dict[item]
```

Comment:
this new function seems out of place in the ` python: move functions from flux-jobs to UtilConfig ` commit.